### PR TITLE
Riese/restrict to UI

### PR DESCRIPTION
**Pull request by MartinRiese**
## Product Description

Add a new column to the organization level table to restrict to which level cases will be synced.

image

## Technical Summary

https://dimagi.atlassian.net/browse/USH-4350

## Feature Flag

None

### Automated test coverage

Added new tests

### QA Plan

Part of larger QA of this feature.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by AddisonDunn — COMMENTED** [review]
Mostly looks good to me

File: corehq/apps/locations/templates/locations/location_types.html [hunk 1]
```diff
@@ -246,10 +246,11 @@ <h2>{% trans "Organization Levels" %}</h2>
               <td>
                 <select class="form-control"
                         data-bind="options: $root.child_loc_types($data),
```
[the remark this answers is not in the document]
Comment (MartinRiese) [reply corehq/apps/locations/templates/locations/location_types.html:248]: You are correct. Not sure what I was thinking about there.

File: corehq/apps/locations/tests/test_views.py [hunk 2]
```diff
@@ -98,3 +99,38 @@ def test_valid_update(self):
...
+    def test_hierarchy(self):
+        loc_type1 = OTHER_DETAILS.copy()
+        loc_type2 = OTHER_DETAILS.copy()
+        loc_type1.update({'name': "new name", 'pk': self.loc_type1.pk, 'code': self.loc_type1.code})
+        loc_type2.update({'name': "new name 2", 'pk': self.loc_type2.pk, 'parent_type': self.loc_type1.pk,
+                          'code': self.loc_type2.code})
+        data = {'loc_types': [loc_type1, loc_type2]}
+        response = self.send_request(data)
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.url, self.url)
```
[the remark this answers is not in the document]
Comment (MartinRiese) [reply corehq/apps/locations/tests/test_views.py:112]: Just to make sure the hierarchy works as I expected.

File: corehq/apps/locations/static/locations/js/location_types.js [hunk 3]
```diff
@@ -33,6 +33,23 @@ hqDefine('locations/js/location_types', [
             });
         };
 
+        self.child_loc_types = function (locType) {
```
[the remark this answers is not in the document]
Comment (MartinRiese) [reply corehq/apps/locations/static/locations/js/location_types.js]: see https://github.com/dimagi/commcare-hq/pull/34427/commits/da90b1f59d0c09534454fc1ca7b08308ded58f1b

**Review by AddisonDunn — DISMISSED** [review]

**Review by AddisonDunn — APPROVED** [review]

**Review by AddisonDunn — APPROVED** [review]

**Pull request merged**